src/common/Compiler.h

```diff
@@ -262,4 +262,19 @@ inline int CountTrailingZeroes(unsigned long long x) { int i = 0; while (i < 64
 	#define UNREACHABLE()
 #endif // UNREACHABLE
 
+/* Use a C++11 braced initializer on MSVC instead of a bracket initializer
+when zeroing a struct. This works around a bug in how MSVC generates implicit
+default constructors. -- Amanieu
+
+The MSVC workaround is known to crash ICC, there is no reason to apply MSVC
+workarounds on non-MSVC compilers. -- illwieckz
+
+I believe GCC, Clang, and MSVC have all been seen to crash during compilation
+on either the () or {} forms of these constructors. -- slipher */
+#if defined(_MSC_VER)
+	#define ZERO_STRUCT( S, T ) do { S.~T(); new( &S ) T{}; } while ( false )
```

Comment:
It should be a template function instead of a macro
```
template<typename T>
void ResetStruct(const T& object) {
... // inside is the same
}
```

Reply:
I tried that:

```c++
template<typename T>
void ResetStruct(const T& object) {
	object.~T();
#if defined(_MSC_VER)
	new( &object ) T{};
#else
	new( &object ) T();
#endif
}
```

I use it that way:

```c++
void CL_ClearState()
{
	ResetStruct( cl );
}
```

But I get that:

```
In file included from src/engine/client/cl_main.cpp:1:
In file included from src/common/Common.h:35:
In file included from src/engine/qcommon/./q_shared.h:161:
src/common/Compiler.h:282:2: error: no matching 'operator new' function for non-allocating placement new expression; include <new>
        new( &object ) T();
        ^~~
src/engine/client/cl_main.cpp:795:2: note: in instantiation of function template specialization 'ResetStruct<clientActive_t>' requested here
        ResetStruct( cl );
        ^
```

I tried to add `#include <new>` but that fixed nothing.

Reply:
Brain fart on my part, there shouldn't be a `const`

Reply:
Thanks, it seems to work. See my additional commit, if that looks OK to you I'll squash.